beacon-chain/core/altair/attestation.go

```diff
@@ -9,3 +21,60 @@ func HasValidatorFlag(flag, flagPosition uint8) bool {
 func AddValidatorFlag(flag, flagPosition uint8) uint8 {
 	return flag | (1 << flagPosition)
 }
+
+// AttestationParticipationFlagIndices retrieves a map of attestation scoring based on Altair's participation flag indices.
+// This is used to facilitate process attestation during state transition and during upgrade to altair state.
+func AttestationParticipationFlagIndices(beaconState state.BeaconStateAltair, data *ethpb.AttestationData, delay types.Slot) (map[uint8]bool, error) {
+	currEpoch := helpers.CurrentEpoch(beaconState)
+	var justifiedCheckpt *ethpb.Checkpoint
+	if data.Target.Epoch == currEpoch {
+		justifiedCheckpt = beaconState.CurrentJustifiedCheckpoint()
+	} else {
+		justifiedCheckpt = beaconState.PreviousJustifiedCheckpoint()
+	}
+
+	matchingSource, matchedTgt, matchedHead, err := MatchingStatus(beaconState, data, justifiedCheckpt)
```

Comment:
So this deviates from the spec here:
```py
    # Matching roots
    is_matching_source = data.source == justified_checkpoint
    is_matching_target = is_matching_source and data.target.root == get_block_root(state, data.target.epoch)
    is_matching_head = is_matching_target and data.beacon_block_root == get_block_root_at_slot(state, data.slot)
    ```
    while we have it as 
    ```go
    	matchedTgt = bytes.Equal(r, data.Target.Root)
    ```
    If there is no reason for divergence, we should keep this as close to the spec as possible.